engine/resources/svc.go

```diff
@@ -425,12 +434,46 @@ func (obj *SvcResAutoEdges) Test(input []bool) bool {
 	return true // keep going
 }
 
-// AutoEdges returns the AutoEdge interface. In this case the systemd units.
+// SvcResAutoEdgesCron holds the state of the svc -> cron auto edge generator.
+type SvcResAutoEdgesCron struct {
+	unit    string // target unit
+	session bool   // user session
+}
+
+// Next returns the next automatic edge.
+func (obj *SvcResAutoEdgesCron) Next() []engine.ResUID {
+	// XXX: should this be true if SvcRes State == "stopped"?
```

Comment:
It definitely works as is for all the cases I can think of.

I've thought about it for a bit, and I'm pretty sure it never needs to be reversed. The only case I wasn't sure about is if pkg uninstalls a svc backing a timer, but that seems fine too. E.g.

```
Install edges:

pkg - svc - cron
```
```
Uninstall edges:

       cron
svc <
       pkg
```

But tests are good, and I'll write a bunch anyway in my next pr.

Reply:
I'll whip up an integration test with this kind of graph soon.

Reply:
drbd?

Actually ask Felix before you do this. I think he has a `mock` thing planned that might help. Not sure.

But ideally we need to write some test cases in pure golang that loop through a resource and the matching effects it would have so that we can define resource behaviour to field properties somewhere...